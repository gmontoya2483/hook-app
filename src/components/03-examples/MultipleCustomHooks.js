import React from 'react';
import './customHooks.css';
import {useFetch} from "../../hooks/useFetch";
import {useCounter} from "../../hooks/useCounter";
// import PropTypes from 'prop-types';

const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);
    const { author, quote } = !!data && data[0];

    return (
        <div>
            <h1>BreakingBad Quotes!!!</h1>
            <hr/>

            {
                (loading)
                    ? (
                        <div className="alert alert-info">
                            Loading ...
                        </div>
                    )
                    : (
                        <blockquote className="blockquote text-end">
                            <p className="mb-3">{ quote }</p>
                            <footer className={'blockquote-footer'}>{ author }</footer>
                        </blockquote>

                    )
            }

            <button
                className={'btn btn-primary'}
                onClick={ () => increment() }
            >
                Siguiente quote
            </button>


        </div>
    );
};

// MultipleCustomHooks.propTypes = {
//
// };

export default MultipleCustomHooks;
