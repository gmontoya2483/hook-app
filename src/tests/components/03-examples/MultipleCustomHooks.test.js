import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";
import {shallow} from "enzyme";
import {useFetch} from "../../../hooks/useFetch";
import {useCounter} from "../../../hooks/useCounter";
jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('pruebas en multiple cuntom hooks', ()=> {


    beforeEach(() => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });

    })




    test('debe de mostrarse correctmante', ()=> {


        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de mostrar la informaciÃ³n', ()=> {

        useFetch.mockReturnValue({
            data: [{
                author: 'Fernando',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBeFalsy();
        expect(wrapper.find('p').text().trim()).toBe('Hola Mundo');
        expect(wrapper.find('footer').text().trim()).toBe('Fernando');


    });

});
