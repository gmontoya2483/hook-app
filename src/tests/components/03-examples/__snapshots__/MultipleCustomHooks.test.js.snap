// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en multiple cuntom hooks debe de mostrarse correctmante 1`] = `
<div>
  <h1>
    BreakingBad Quotes!!!
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    Loading ...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
